diagnostic.js

```diff
@@ -32,7 +42,20 @@ db.dropDatabase();
 // Jon Arryn
 
 // your code begins here
-
+  db.houses.update(
+    { name: "Stark" },
+    { members: ["Ned Stark", "Arya Stark", "Sansa Stark"]}
+  );
+
+  db.houses.insert({
+    name: "Targaryen",
+    members: ["Viserys Targaryen", "Daenerys Targaryen"]
+  });
+
+  db.houses.update(
+    { name: "Arryn" },
+    { $push: { members: "Jon Arryn", } }
+  );
```

Comment:
a few issues here:
1) `.insert` is for adding, not updating
2) recall that `.update` requires 2 arguments - the 1st is thing that we are searching with, and the 2nd is the thing that we are replacing it with. You used $push only on one of them, which is correct, but not the others, which means that you have replaced the existing data in that object with the `member` object.